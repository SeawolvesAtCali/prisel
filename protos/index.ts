export * as monopolypb from './gen_ts/monopoly/index';
export * as priselpb from './gen_ts/prisel/index';
export * as tic_tac_toepb from './gen_ts/tic-tac-toe/index';
