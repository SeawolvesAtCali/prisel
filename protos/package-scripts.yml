scripts:
    clean: rm -rf gen_ts gen_csharp && mkdir gen_ts gen_csharp
    protoc:
        protoc --experimental_allow_proto3_optional --ts_opt force_client_none --ts_out gen_ts
        --csharp_opt base_namespace= --csharp_out gen_csharp --proto_path . $(find . -type f -name
        "*.proto")
    ts-check: tsc
    barrel: barrelsby -c barrelsby.json --delete
    rollup: rollup --config --silent
