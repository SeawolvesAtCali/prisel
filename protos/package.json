{
    "name": "@prisel/protos",
    "version": "0.1.3",
    "description": "",
    "license": "MIT",
    "author": "",
    "main": "./lib/index.cjs.js",
    "module": "./lib/index.esm.js",
    "browser": "./lib/index.umd.js",
    "types": "./dist/index.d.ts",
    "scripts": {
        "build": "nps barrel ts-check rollup",
        "test": "echo \"Error: no test specified\""
    },
    "dependencies": {
        "flatbuffers": "^2.0.3",
        "@types/long": "^4.0.0",
        "buffer": "^5.6.0",
        "long": "^4.0.0"
    },
    "devDependencies": {
        "@prisel/configs": "1.0.0",
        "barrelsby": "^2.2.0",
        "nps": "^5.9.12",
        "rollup": "^2.7.6",
        "tslib": "^2.0.0",
        "typescript": "^4.0.5"
    }
}
