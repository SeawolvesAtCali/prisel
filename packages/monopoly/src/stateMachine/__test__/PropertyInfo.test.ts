import { monopolypb, protobuf } from '@prisel/protos';

describe('PropertyInfo', () => {
    test('PropertyInfo contains currentLevel', () => {
        const promptPurchase: monopolypb.PromptPurchaseRequest = {
            property: {
                name: 'name',
                currentLevel: 0,
                cost: 100,
                rent: 10,
            },
            levels: [],
            isUpgrade: false,
            moneyAfterPurchase: 1000,
        };
        const packed = protobuf.any.Any.pack(promptPurchase, monopolypb.PromptPurchaseRequest);
        const unpacked = protobuf.any.Any.unpack(packed, monopolypb.PromptPurchaseRequest);
        expect(unpacked).toEqual(promptPurchase);
        // toJson removes field that matches default value, so currentLevel will
        // not appear.
        expect(
            JSON.stringify(
                protobuf.any.Any.toJson(packed, {
                    typeRegistry: [monopolypb.PromptPurchaseRequest],
                }),
                null,
                2,
            ),
        ).toMatchSnapshot();
    });
});
