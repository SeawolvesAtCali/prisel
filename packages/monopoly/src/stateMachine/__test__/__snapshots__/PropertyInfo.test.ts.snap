// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyInfo PropertyInfo contains currentLevel 1`] = `
"{
  \\"property\\": {
    \\"cost\\": 100,
    \\"rent\\": 10,
    \\"name\\": \\"name\\"
  },
  \\"moneyAfterPurchase\\": 1000,
  \\"@type\\": \\"type.googleapis.com/monopoly.PromptPurchaseRequest\\"
}"
`;
